Date: Dec 14, 2022
From: Aravindan NarasimhapuramChakravarthy <aravindan_narasimhapuramchakravarthy@comcast.com>
Source: COMCAST
Subject: Build bind without readline config since readline-5.2 is casuing build errors in bind.
Re-create version specific patches to fix patch/install errors.
Signed-off-by: Aravindan NarasimhapuramChakravarthy <aravindan_narasimhapuramchakravarthy@comcast.com>
Index: bind-9.18.5/bin/named/server.c
===================================================================
--- bind-9.18.5.orig/bin/named/server.c
+++ bind-9.18.5/bin/named/server.c
@@ -7613,7 +7613,7 @@ generate_session_key(const char *filenam
 					NULL, now, now, mctx, NULL, &tsigkey));
 
 	/* Dump the key to the key file. */
-	fp = named_os_openfile(filename, S_IRUSR | S_IWUSR, first_time);
+	fp = named_os_openfile(filename, S_IRUSR, first_time);
 	if (fp == NULL) {
 		isc_log_write(named_g_lctx, NAMED_LOGCATEGORY_GENERAL,
 			      NAMED_LOGMODULE_SERVER, ISC_LOG_ERROR,
@@ -7625,9 +7625,7 @@ generate_session_key(const char *filenam
 	fprintf(fp,
 		"key \"%s\" {\n"
 		"\talgorithm %s;\n"
-		"\tsecret \"%.*s\";\n};\n",
-		keynamestr, algstr, (int)isc_buffer_usedlength(&key_txtbuffer),
-		(char *)isc_buffer_base(&key_txtbuffer));
+		"\n};\n", keynamestr, algstr);
 
 	CHECK(isc_stdio_flush(fp));
 	result = isc_stdio_close(fp);
